bark/generation.py

```diff
@@ -129,13 +118,25 @@ def _cast_bool_env_var(s):
     )
 
 
+import torch
+
+if hasattr(torch.nn.functional, 'flash_attention'):
+    print("------------------------------------------------->Flash Attention is available in PyTorch.")
+    flash_attention_available = True
+else:
+    # print("------------------------------------------------->Flash Attention is NOT available in PyTorch.")
+    flash_attention_available = False
+
+
+
+
 def _grab_best_device(use_gpu=True):
     if torch.cuda.device_count() > 0 and use_gpu:
         device = "cuda"
     elif torch.backends.mps.is_available() and use_gpu and GLOBAL_ENABLE_MPS:
         device = "mps"
     else:
-        device = "cpu"
+        device = "cuda"
```

Comment:
```suggestion
        device = "cpu"
```

Users can choose CPU or GPU, this code breaks the ability for users to choose.

Reply:
Again! Thank you for pointing this out. Upon reflection, I realize this modification was likely an oversight from my side. During development, my primary focus was on optimizing for CUDA, and since I was working with an H100 GPU, I hardcoded "cuda" to ensure all tests leveraged GPU resources. This change inadvertently removed the flexibility for users to select their preferred device.

I agree this behavior limits the usability of the function, especially for non-GPU setups or cases where users want to test on CPU or MPS. I'll update the code to respect user choice while ensuring that CUDA optimizations are used only when a GPU is explicitly selected.

Thanks again for bringing this to my attention!